reference/src/abort-and-assert/clever-errors.md

```diff
@@ -0,0 +1,230 @@
+# Clever Errors
+
+Clever errors are a feature that allows for more informative error messages when an assertion fails
+or an abort is raised. They are a source feature and compile to a `u64` abort code value that
+contains the information needed to access the line number, constant name, and constant value given
+the clever error code and the module that the clever error constant was declared in. Because of
+this, post-processing is required to go from the `u64` abort code value to a human-readable error
+message. This post-processing is automatically performed by the Sui GraphQL server, as well as the
+Sui CLI. If you want to manually decode a clever abort code, you can use the process outlined in
+[Inflating Clever Abort Codes](#inflating-clever-abort-codes) to do so.
+
+> Clever errors include source line information amongst other data. Because of this their value may
+> change due to any changes in the source file (e.g., due to auto-formatting, adding a new module
+> member, or adding a newline).
+
+## Clever Abort Codes
```

Comment:
Can you add a link to a matching page in the `book/`?

Reply:
We don't have one in the `book/` (at least not yet) so not really anything to reference here unless I'm misunderstanding something.